server/src/test/java/org/cloudfoundry/identity/uaa/provider/IdentityProviderEndpointsTest.java

```diff
@@ -437,6 +722,47 @@ void testDeleteIdentityProviderExisting() {
         assertEquals(validIDP, deleteResponse.getBody());
     }
 
+    @Test
+    void testDeleteIdentityProviderMirrored() {
```

Comment:
What does mirrored mean here? Is a duplicate 100% redundant idp? If so, shouldn't that be protected against being created? If mirrored is something else, what does it mean?

Reply:
For an IdP, its mirrored IdP is the copy of it in another identity zone (see the diagram in issue #2505). The "original" IdP and the mirrored IdP are identical except for the fields `id`, `zid`, `alias_id` and `alias_zid`. In the original IdP, the alias fields reference its mirrored IdP and vice versa.

Reply:
My confusion lies in the term mirror. that implies it is kept in sync and should remain a "mirror" or identical to the original.  Though in this case, it is a copy, which can over time diverge and become different, especially in regards to permissions, yes?

Reply:
The two IdPs are actually kept in sync. Any property change to one of them is propagated to the other, respectively.

The sentence you mentioned refers to us copying the IdPs (or later also users) instead of using "pointers" to IdPs in the corresponding other identity zone. This way, we do not need to add logic for resolving these pointers whenever an operation on an IdP is performed. Instead, the existing functionalities work out-of-the-box for the mirrored and original IdP, as both of them are - apart from their `aliasId` and `aliasZid` properties - similar to any other IdP. We just need to pay attention that the changes are propagated to the mirrored user during CRUD operations.